### ISSUE TITLE
[Feature]: A developer-signed, non-debuggable variant of Termux on GitHub Releases

### ISSUE THREAD

**Issue by ghost** [termux/termux-app#4094]
### Feature description

There should be a developer-signed, non-debuggable variant of Termux published on GitHub Releases. GitHub Releases is the fastest way to get new updates since you don't have to go through Google Play's or F-Droid's update review processes. You can keep the Termux app up-to-date by manually checking for updates or using an app to automatically check for updates/update it for you.

### Additional information

Most Android developers using GitHub Releases publish developer-signed, non-debuggable apps.

**Comment by agnostic-apollo** [comment]
There are plans to publish another source on our site after/with next main release, a key was already generated. Publishing both debug and release builds on github would be problematic, there are already separate android 5/6 and 7+ variants and users have been blindly downloading the android 5/6 variants, as its download count is much higher than the android 7+ ones, which it should not be, will fix that in next beta. We may eventually need to release different execution type/targetsdk variants too.

Playstore having removed `sharedUserId` without warning users if a plugin is installed from a different source has already created a mess, as signatures don't need to be same for main termux app and plugins.

So ultimately situation would get worse if another key is added, even though we may need to.

**Comment by agnostic-apollo** [comment]
Yeah, I already have written docs for it, but they aren't complete yet due to some blockers.

**Comment by agnostic-apollo** [comment]
I have manually added a downloads section to the release info, that should help. On mobile site, github does not show full filename of artifacts as its too long, so that might be part of the problem as users would not be seeing the `android-*` and architecture tags at the end. The release artifacts default sort order can't be changed either to list universal architecture apk first. The downloads section shows the full name and lists universal apk first to solve those issues. Will have to automate that with github workflow as manually creating the section is cumbersome.

https://github.com/termux/termux-app/releases/tag/v0.119.0-beta.1

**Comment by agnostic-apollo** [comment]
`v0.119.0` will be the next main release that is currently going through betas.

**Comment by ghost** [comment]
Why not maybe add developer-signed, non-debuggable releases to future releases on GitHub Releases and have testkey-signed, debuggable releases on GitHub Actions?

**Comment by agnostic-apollo** [comment]
Users using github releases will not be able to update from github releases if keys are changed.

**Comment by ghost** [comment]
> Users using github releases will not be able to update from github releases if keys are changed.

If that were to change then as an upgrade plan I would recommend telling the user that the update requires a Termux reinstallation. It should also let the user know how to backup Termux: https://wiki.termux.com/wiki/Backing_up_Termux.

**Comment by agnostic-apollo** [comment]
We do not want to deal with that mess and is not a good idea to suddenly change keys. A new install source will be provided on our site with our own private key in future.
